ts/src/pro/coinex.ts

```diff
@@ -1073,58 +1126,33 @@ export default class coinex extends coinexRest {
         const spotMessageHash = 'authenticated:spot';
         const swapMessageHash = 'authenticated:swap';
         const messageHash = isSpot ? spotMessageHash : swapMessageHash;
         const future = client.future (messageHash);
         const authenticated = this.safeValue (client.subscriptions, messageHash);
-        if (type === 'spot') {
-            if (authenticated !== undefined) {
-                return await future;
-            }
-            const requestId = this.requestId ();
-            const subscribe: Dict = {
-                'id': requestId,
-                'future': spotMessageHash,
-            };
-            const signData = 'access_id=' + this.apiKey + '&tonce=' + this.numberToString (time) + '&secret_key=' + this.secret;
-            const hash = this.hash (this.encode (signData), md5);
-            const request: Dict = {
-                'method': 'server.sign',
-                'params': [
-                    this.apiKey,
-                    hash.toUpperCase (),
-                    time,
-                ],
-                'id': requestId,
-            };
-            this.watch (url, messageHash, request, requestId, subscribe);
-            client.subscriptions[messageHash] = true;
-            return await future;
-        } else {
-            if (authenticated !== undefined) {
-                return await future;
-            }
-            const requestId = this.requestId ();
-            const subscribe: Dict = {
-                'id': requestId,
-                'future': swapMessageHash,
-            };
-            const signData = 'access_id=' + this.apiKey + '&timestamp=' + this.numberToString (time) + '&secret_key=' + this.secret;
-            const hash = this.hash (this.encode (signData), sha256, 'hex');
-            const request: Dict = {
-                'method': 'server.sign',
-                'params': [
-                    this.apiKey,
-                    hash.toLowerCase (),
-                    time,
-                ],
-                'id': requestId,
-            };
-            this.watch (url, messageHash, request, requestId, subscribe);
-            client.subscriptions[messageHash] = true;
+        if (authenticated !== undefined) {
             return await future;
         }
+        const requestId = this.requestId ();
+        const subscribe: Dict = {
+            'id': requestId,
+            'future': spotMessageHash,
+        };
+        const hmac = this.hmac (this.encode (timestamp), this.encode (this.secret), sha256, 'hex');
+        const request: Dict = {
+            'id': requestId,
+            'method': 'server.sign',
+            'params': {
+                'access_id': this.apiKey,
+                'signed_str': hmac.toLowerCase (),
+                'timestamp': time,
+            },
+        };
+        this.watch (url, messageHash, request, requestId, subscribe);
+        client.subscriptions[messageHash] = true;
+        return await future;
```

Comment:
as discussed earlier, we'r emissing `handleErrors`,
some of the error messages we have to parse

```
2024-08-06T13:52:44.557Z onMessage { id: 1, code: 20001, message: 'invalid argument' }
2024-08-06T13:52:44.558Z sending { method: 'order.subscribe', params: { market_list: [] }, id: 2 }
2024-08-06T13:52:44.894Z onMessage { id: 2, code: 21001, message: 'require auth' }
```